### PR TITLE
Add assistant capabilities to control rendering components

### PR DESCRIPTION
**Pull request by songkant-aws**
### Description
Add assistant feature capabilities to allow users to override capability switcher function to control whether to show assistant feature UI components.

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test.
- [x] New functionality has user manual doc added.
- [x] Commits are signed per the DCO using --signoff.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: common/types/config.ts [hunk 1]
```diff
@@ -6,8 +6,7 @@
 import { schema, TypeOf } from '@osd/config-schema';
 
 export const configSchema = schema.object({
-  // TODO: add here to prevent this plugin from being loaded
-  // enabled: schema.boolean({ defaultValue: true }),
+  enabled: schema.boolean({ defaultValue: true }),
```
Comment (Hailong-am) [inline common/types/config.ts:9]: enable this will allow whole dashboard assistant been disabled, is this expected?

File: public/plugin.tsx [hunk 2]
```diff
@@ -146,12 +160,18 @@ export class AssistantPlugin
...
+          if (isSubscriptionActive) {
+            const { renderText2VizApp } = await import('./text2viz');
+            return renderText2VizApp(params, {
+              ...coreStart,
+              ...pluginsStart,
+              setHeaderActionMenu: params.setHeaderActionMenu,
+            });
+          } else {
+            return () => {};
+          }
```
Comment (Hailong-am) [inline public/plugin.tsx:174]: do we want to render some tips or instructions to direct user to active the subscription?

**Review by Hailong-am — APPROVED** [review]

File: public/plugin.tsx (hunk 2 above)
Comment (songkant-aws) [reply public/plugin.tsx]: Add a default "Application Not Activated" danger tips now

File: common/types/config.ts (hunk 1 above)
Comment (songkant-aws) [reply common/types/config.ts:9]: Yes, I think it's upon agreement. Is it true? @ruanyl

Comment (ruanyl) [reply common/types/config.ts:9]: I think so, this flag allows us to turnoff the entire plugin

File: server/plugin.ts [hunk 3]
```diff
@@ -61,16 +62,17 @@ export class AssistantPlugin implements Plugin<AssistantPluginSetup, AssistantPl
...
-        show: true,
-      },
-    }));
+    // Register router for alert insight
+    if (config.alertInsight.enabled) {
+      registerSummaryAssistantRoutes(router, assistantServiceSetup);
+    }
+
+    core.capabilities.registerProvider(capabilitiesProvider);
+    core.capabilities.registerSwitcher(capabilitiesSwitcher);
```
[the remark this answers is not in the document]
Comment (songkant-aws) [reply server/plugin.ts:75]: Sure, that also makes sense. Developers can register their own switch per requirement.

**Review by ruanyl — APPROVED** [review]

**Comment by ruanyl** [comment]
@songkant-aws backport to 2.x failed.

**Comment by songkant-aws** [comment]
> @songkant-aws backport to 2.x failed.

Ok, will fix it. The PR is merged after this one [#266 ]. Since some conflicts are resolved based on #266, I will wait for its backport. Let me ping the owner to backport #266.

**Pull request merged**